src/arrayMethodSort.js

```diff
@@ -3,10 +3,47 @@
 /**
  * Implement method Sort
  */
+
+const compareAsStrings = (a, b) => {
+  if (String(a) > String(b)) {
+    return 1;
+  }
+
+  if (String(a) === String(b)) {
+    return 0;
+  }
+
+  return -1;
+};
+
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
+  [].__proto__.sort2 = function(compareFunction = compareAsStrings) {
     // write code here
+
+    let changeCount = this.length;
+    let change = 0;
+    let difArray = [...this];
```

Comment:
this array looks redundant, you can just create one temporary variable (btw, this var also is not required, you can implement this function without it)

Reply:
Ok, removed unnecessary elements